src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -109,20 +119,18 @@ public bool SnippetShortcutExists_NonBlocking(string shortcut)
         public virtual bool ShouldFormatSnippet(SnippetInfo snippetInfo)
             => false;
 
-        private void PopulateSnippetCaches()
+        private async Task PopulateSnippetCachesAsync()
         {
             Debug.Assert(_expansionManager != null);
-
-            var token = _waiter.BeginAsyncOperation(GetType().Name + ".Start");
+            AssertIsForeground();
```

Comment:
```suggestion
            await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Reply:
Does this actually just need to get deleted? It looks like we're going to the foreground to do a cast but then immediately jumps to the background thread to use it? My hunch is this is a managed object so the UI thread jump isn't necessary at all.

Reply:
Yup, per DavidP this should be a machine translated version - so casting should be callable from any thread, but any more complex stuff may not be.

Reply:
@jasonmalinowski If it's not _documented_ as a managed object, then it's not safe to assume it is.